carbon/datapoint.go

```diff
@@ -0,0 +1,57 @@
+package carbon
+
+import "strconv"
+
+const (
+	nextPart      = " "
+	nextNode      = "."
+	nextTag       = ";"
+	nextTagValue  = "="
+	nextDatapoint = "\n"
+)
+
+// Datapoint is a value stored at a timestamp bucket.
+// If no value is recorded at a particular timestamp bucket in a series,
+// the value will be None (null).
+// Doc: https://graphite.readthedocs.io/en/latest/terminology.html
+type Datapoint struct {
+	path      string
```

Comment:
I would suggest to use []bytes or bytes.Buffer here instead. Main reason is that currently `WithTag`, `WithPrefix` methods would cause a lot of unneeded allocations (strings in golang are immutable).

Reply:
Yeah, you're right! The optimization won't be premature this time